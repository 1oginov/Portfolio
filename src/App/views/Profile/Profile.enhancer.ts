import { connect } from 'react-redux';

import State from 'State';

import { Props } from './Profile';

const mapStateToProps = ({ firebase: { firebase: { profile } } }: State): Props => ({ profile });

export default connect<Props, {}, {}, State>(mapStateToProps);
