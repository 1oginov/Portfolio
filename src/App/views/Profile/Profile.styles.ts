import { withStyles } from '@material-ui/core';

export default withStyles(theme => ({
  avatar: {
    height: 128,
    marginBottom: theme.spacing(3),
    marginTop: theme.spacing(3),
    width: 128,
  },
  email: {
    //
  },
  headline: {
    marginBottom: theme.spacing(3),
    marginTop: theme.spacing(3),
    textAlign: 'center',
  },
  name: {
    //
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    padding: theme.spacing(6),
  },
  signOut: {
    marginBottom: theme.spacing(3),
    marginTop: theme.spacing(3),
  },
}));
