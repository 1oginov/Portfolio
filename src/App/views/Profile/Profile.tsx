import { Avatar, Typography } from '@material-ui/core';
import * as React from 'react';

import { ProfileState } from 'Firebase/State';
import { ResetLogosAction, ResetQuotesAction } from 'Stocks/actions';
import LogoutButton from 'User/components/LogoutButton';

interface Props {
  classes: { [key: string]: string };
  profile: ProfileState;
  resetLogos: ResetLogosAction;
  resetQuotes: ResetQuotesAction;
}

const Profile: React.FunctionComponent<Props> = ({
  classes, profile, resetLogos, resetQuotes,
}: Props) => (
  <div className={classes.root}>

    <Typography className={classes.headline} variant="h4">Profile</Typography>

    {profile.avatarUrl && (
      <Avatar className={classes.avatar} src={profile.avatarUrl} />
    )}

    {profile.displayName && (
      <Typography variant="h6">{profile.displayName}</Typography>
    )}

    {profile.email && (
      <Typography>{profile.email}</Typography>
    )}

    <LogoutButton
      className={classes.signOut}
      onLogout={() => {
        resetLogos();
        resetQuotes();
      }}
    >
      Sign Out
    </LogoutButton>

  </div>
);

export default Profile;
