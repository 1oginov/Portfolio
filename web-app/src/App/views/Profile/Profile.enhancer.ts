import { connect } from 'react-redux';

import { resetVibrantPalettes, ResetVibrantPalettesAction } from 'Firebase/actions';
import { ProfileState } from 'Firebase/State';
import State from 'State';
import {
  resetLogos, ResetLogosAction, resetQuotes, ResetQuotesAction,
} from 'Stocks/actions';

interface StateProps {
  profile: ProfileState;
}

interface DispatchProps {
  resetLogos: ResetLogosAction;
  resetQuotes: ResetQuotesAction;
  resetVibrantPalettes: ResetVibrantPalettesAction;
}

const mapStateToProps = ({ firebase: { firebase: { profile } } }: State): StateProps => ({ profile });

const mapDispatchToProps = { resetLogos, resetQuotes, resetVibrantPalettes };

export default connect<StateProps, DispatchProps, Record<string, never>, State>(mapStateToProps, mapDispatchToProps);
