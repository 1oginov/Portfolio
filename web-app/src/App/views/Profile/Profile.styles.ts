import { withStyles } from '@material-ui/core';

import { viewHeadline, viewRoot } from 'Styles/mixins';

export default withStyles((theme) => ({
  avatar: {
    height: 128,
    marginBottom: theme.spacing(3),
    marginTop: theme.spacing(3),
    width: 128,
  },
  headline: viewHeadline(theme),
  root: {
    ...viewRoot(theme),
    justifyContent: 'center',
  },
  signOut: {
    marginBottom: theme.spacing(3),
    marginTop: theme.spacing(3),
  },
}));
