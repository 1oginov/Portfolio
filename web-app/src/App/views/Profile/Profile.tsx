import { Avatar, Typography } from '@material-ui/core';
import * as React from 'react';

import {
  ResetVibrantPalettesAction,
} from 'Layers/Application/ActionCreators/FirebaseActionCreators/FirebaseActionCreators';
import { ProfileState } from 'Layers/Application/States/FirebaseState/FirebaseState';
import { LogoutButton } from 'Layers/Presentation/Components/LogoutButton';
import { ResetLogosAction, ResetQuotesAction } from 'Stocks/actions';

interface Props {
  classes: { [key: string]: string };
  profile: ProfileState;
  resetLogos: ResetLogosAction;
  resetQuotes: ResetQuotesAction;
  resetVibrantPalettes: ResetVibrantPalettesAction;
}

const Profile: React.FunctionComponent<Props> = ({
  classes, profile, resetLogos, resetQuotes, resetVibrantPalettes,
}: Props) => (
  <div className={classes.root}>

    <Typography className={classes.headline} variant="h4">Profile</Typography>

    {profile.avatarUrl && (
      // TODO: rel="no-referrer" should be passed to the Image object created in background to download the image and
      //  avoid 403. Pending fix from the Material UI library.
      <Avatar className={classes.avatar} src={profile.avatarUrl} />
    )}

    {profile.displayName && (
      <Typography variant="h6">{profile.displayName}</Typography>
    )}

    {profile.email && (
      <Typography>{profile.email}</Typography>
    )}

    <LogoutButton
      className={classes.signOut}
      onLogout={() => {
        resetLogos();
        resetQuotes();
        resetVibrantPalettes();
      }}
    >
      Sign Out
    </LogoutButton>

  </div>
);

export default Profile;
