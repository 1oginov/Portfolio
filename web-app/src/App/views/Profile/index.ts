import Profile from './Profile';
import enhance from './Profile.enhancer';
import style from './Profile.styles';

export default style(enhance(Profile));
