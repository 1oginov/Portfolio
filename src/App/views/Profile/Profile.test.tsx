import { shallow } from 'enzyme';
import * as React from 'react';

import Profile from './Profile';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Profile
      profile={{
        avatarUrl: 'http://example.com/avatar.png',
        displayName: 'Dolph',
        email: 'dolph@example.com',
        isEmpty: false,
        isLoaded: true,
      }}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
