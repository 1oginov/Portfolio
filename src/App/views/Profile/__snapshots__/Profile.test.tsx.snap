// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h1>
    Profile
  </h1>
  <div>
    <img
      alt="Avatar"
      src="http://example.com/avatar.png"
    />
    <strong>
      Dolph
    </strong>
    <em>
      dolph@example.com
    </em>
  </div>
  <div>
    <withFirebase(withHandlers(LogoutButton))>
      Logout
    </withFirebase(withHandlers(LogoutButton))>
  </div>
</Fragment>
`;
